### DISCUSSION TITLE
Page coordinates are offset from PDF

### DISCUSSION THREAD

**Discussion by brandxntu** [pymupdf/PyMuPDF#1157] [Looking for help] [answered]
I am attempting to draw some objects onto a PDF file, however the results are offset from their original positions and shrunk down.

For context, the PDF file I am using is 2448x1584 pixels. When drawing the Rect(0,0,2448,1584) (which is the result of page.rect) onto the page, you would expect it to fill the entirety of the page as seen in the left image, but I instead get a result similar to the right image where only one corner of the page is filled. 

![image](https://user-images.githubusercontent.com/20878486/126568876-56a5401b-a73a-4733-a816-6d070b01b6d1.png)

Has anyone experienced this issue before? Would this be an issue with the PDF itself?

**Comment by JorjMcKie** [comment, marked as answer]
> Has anyone experienced this issue before? Would this be an issue with the PDF itself?

Fairly probably, this is caused by a situation handled [here](https://pymupdf.readthedocs.io/en/latest/faq.html#misplaced-item-insertions-on-pdf-pages).
Execute `page.clean_contents()` before inserting anything.
